### DISCUSSION TITLE
Compatibility Level is read only?

### DISCUSSION THREAD

**Discussion by otykier** [TabularEditor/TabularEditor3#104] [🙋♀️TE3 Q&A] [answered]
See question below.

**Comment by mshannon4** [comment]
I am trying to deploy a Power BI desktop model using Tabular Editor 3 to my SSAS server.  I get an error - The specified server does not support Compatibility Level 1550.   So I attempt to lower the compatibility level of the model to 1500 (server's level).  The option is read only.  Why is this Compatibility Level setting read only.  Additionally, 1550 is not an option for SSAS (https://docs.microsoft.com/en-us/analysis-services/tabular-models/compatibility-level-for-tabular-models-in-analysis-services?view=asallproducts-allversions).   This has been plaguing me for a while.

**Reply by otykier** [reply, marked as answer]
Migrating from Power BI to SSAS is not straightforward. There are a couple of things to be aware about:

1) Tabular Editor restricts the types of changes you can make to Power BI Desktop model by default, to ensure we conform with the [supported write operations for Power BI Desktop external tools](https://docs.microsoft.com/en-us/power-bi/transform-model/desktop-external-tools). This explains why the Compatibility Level property is grayed out. To workaround this, start by saving your Power BI Desktop model as a .bim file using Tabular Editor, then load that .bim file in TE. Now, you should be able to change the Compatibility Level (maybe!)
2) Compatibility Level downgrades are not always possible, if your model uses features that are not available at lower compatibility levels. One example is [aggregations](https://docs.microsoft.com/en-us/dotnet/api/microsoft.analysisservices.tabular.column.alternateof?view=analysisservices-dotnet). If your model uses aggregations, you can't go below compatibility level 1460 before you remove all aggregations from your model. There are many other such restrictions. You must go through Microsoft docs to find out which restrictions apply.
3) While you're at it, remember to change the Compatibility Mode from "Power BI" to "Analysis Services".
4) Power BI Desktop models do not define explicit data sources in the TOM. Instead, each data source is defind within each partition M query, and Power BI Desktop manages the credentials. In order to deploy to SSAS, you will have to create an explicit data source in the model, for each data source you're using, and then update each table partition M query to reference that data source. In this regard, be aware that Power BI supports **a lot** of data sources, that are simply not supported in SSAS - mostly exotic stuff like Databricks, Parquet and so on.
5) Even if you manage to downgrade the compatibility level and specify explicit data sources, there are still no guarantees that you can deploy and process the model on SSAS without issues. It's a trial and error process really. Many times, you will have more success by creating a new model from scratch in SSAS, import the tables, etc. that you need from the source, and then migrate calculated tables, measures, etc. from your Power BI Desktop model, simply by copying from one to the other using Tabular Editor.

Good luck!

**Comment by mshannon4** [comment]
Daniel
Thank you for the response, it was very helpful and very thorough.
The 1550 compatibility issue was nagging me.

One more question:  We use Visual Studio 2019 as another way to mange our tabular models along with Tabular Editor (free version) and desktop PBI.
We looking at Tabular Editor 3.   Why would we need Tabular editor 3 when we have the tools mentioned above?

Thanks for all the work you do

CHEERS
Michael Shannon
Senior Data Integration Specialist
512 -970-4943

**Reply by otykier** [reply]
Here is an overview of the features added in TE3: https://docs.tabulareditor.com/#choosing-between-te2-and-te3

**Comment by mshannon4** [comment]
THANK YOU!!!

CHEERS
Michael Shannon
Senior Data Integration Specialist
512 -970-4943
